src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -517,128 +505,116 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
             self.widgets[widx::closeButton].type = WidgetType::none;
         }
 
-        // Update the title.
+        const auto& currentTab = kMainTabInfo[self.currentTab];
+        const auto& subTabs = currentTab.subTabs;
+        const bool showSecondaryTabs = !subTabs.empty() && FilterLevel(self.var_856) != FilterLevel::beginner;
+
+        // Update page title
         auto args = FormatArguments(self.widgets[widx::caption].textArgs);
-        args.push(_tabDisplayInfo[self.currentTab].name);
+        if (showSecondaryTabs)
+            args.push(subTabs[self.currentSecondaryTab].name);
+        else
+            args.push(kMainTabInfo[self.currentTab].name);
 
-        const auto& tabFlags = _tabDisplayInfo[self.currentTab].flags;
-        const bool showSecondaryTabs = (tabFlags & ObjectTabFlags::filterByVehicleType) != ObjectTabFlags::none;
-        for (int i = widx::vehicleTypeTrain; i <= widx::vehicleTypeShip; i++)
+        // Toggle secondary tabs
+        for (auto i = 0U; i < 8U; i++)
```

Comment:
what is this `8`? is it always 8 secondarytabs?

Reply:
There are currently only 8 widgets defined for secondary tabs. Adding a constant would be good, yes.